ios/RNSScreenStackHeaderConfig.mm

```diff
@@ -714,9 +714,13 @@ + (void)updateViewController:(UIViewController *)vc
 #if defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && defined(__IPHONE_26_0) && \
     __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_26_0
           if (@available(iOS 26.0, *)) {
-            // Workaround for missing search bar on root stack screen.
-            // See: https://github.com/software-mansion/react-native-screens/pull/3098
-            navitem.searchBarPlacementAllowsToolbarIntegration = NO;
+            if (navitem.preferredSearchBarPlacement != UINavigationItemSearchBarPlacementStacked) {
```

Comment:
We need some commments why this is implemented like that. Exactly what you told me on call 🙏🏻